kopf/_cogs/structs/patches.py

```diff
@@ -58,11 +58,12 @@ def __init__(self, __src: "Patch") -> None:
 # Event-handling structures, used internally in the framework and handlers only.
 class Patch(Dict[str, Any]):
 
-    def __init__(self, __src: Optional[MutableMapping[str, Any]] = None) -> None:
+    def __init__(self, __src: Optional[MutableMapping[str, Any]] = None, body = {}) -> None:
```

Comment:
Ugh. I missed this. Sorry. The strict typing requires a type here (see the [CI errors](https://github.com/nolar/kopf/pull/803/checks?check_run_id=3131928690)).

```python
    def __init__(
        self,
        __src: Optional[MutableMapping[str, Any]] = None,
        **,
        body: Optional[bodies.RawBody] = None,
    ) -> None:
```

Also above (in this alphabetical order, to satisfy isort/pre-commit):

```python
from kopf._cogs.structs import bodies, dicts
```

I guess, this will be enough. Please check with `mypy --strict kopf` (or via the CI).

Reply:
PS: Maybe `# type: ignore` is needed below on the line where you select a key from it (`_search[key]`). This would be fine, as it is explicitly caught there with TypeError.

Reply:
Updated and verified with mypy